src/Command/GenerateObjectAclCommand.php

```diff
@@ -30,6 +32,24 @@ class GenerateObjectAclCommand extends QuestionableCommand
      */
     protected $userEntityClass = '';
 
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    /**
+     * @var RegistryInterface
+     */
+    private $em;
```

Comment:
Why not $registry v

Reply:
Do you know how can replace this service injection? This is referencing "doctrine" service, which is provided by the not required dependency "doctrine/doctrine-bundle":

https://github.com/sonata-project/SonataAdminBundle/blob/64b83e290566dc609a7a30d6a419c1359430dfc9/src/Command/GenerateObjectAclCommand.php#L116

I was thinking to inject the available model managers instead, and obtain the corresponding registry from them, but that approach will require to update `ModelManagerInterface` in order to expose the registry or at least the return value of `Registry::getEntityNamespace()` method. This will also force the user to indicate model manager when invoking the command, at least if there is more than one.

Otherwise, could you provide an alternative solution to fix the tests from `AddDependencyCallsCompilerPassTest`?

Reply:
> at least the return value of Registry::getEntityNamespace() method

A method named `getModelNamespace()` might make sense? You are not forced to add it to the `ModelManagerInterface`, it could be added to a separate interface and the command would throw if the model manager does not implement it.

Another, perhaps more short term solution is to conditionally inject the `doctrine` service if is present, and throw if someone attempts to use the command but `doctrine` hasn't been injected. That way the dependency can stay optional.

Reply:
Thank you so much for the hint!